### PR TITLE
Handle decimal type in arrow schema

### PR DESCRIPTION
**Pull request by fvaleye**
Handle decimal type on a delta table schema #91

### CODE REVIEW

File: rust/src/delta_arrow.rs [hunk 1]
```diff
@@ -47,8 +49,12 @@ impl From<&schema::SchemaDataType> for ArrowDataType {
                     "byte" => ArrowDataType::Int8,
                     "float" => ArrowDataType::Float32,
                     "double" => ArrowDataType::Float64,
-                    "boolean" => ArrowDataType::Boolean,
```
Comment (nevi-me) [inline rust/src/delta_arrow.rs:50]: Why does your change remove "boolean" and "binary"?

Comment (fvaleye) [reply rust/src/delta_arrow.rs:50]: Oh, good catch! We don't want to change this :) I fix it.

File: rust/src/delta_arrow.rs [hunk 2]
```diff
@@ -39,6 +40,7 @@ impl From<&schema::SchemaDataType> for ArrowDataType {
     fn from(t: &schema::SchemaDataType) -> Self {
         match t {
             schema::SchemaDataType::primitive(p) => {
+                let decimal_regex = Regex::new(r"\((\d{1,2}),(\d{1,2})\)").unwrap();
```
Comment (houqp) [inline rust/src/delta_arrow.rs:43]: we should wrap this with lazy static to repetitive compilation: https://docs.rs/regex/1.4.3/regex/#example-avoid-compiling-the-same-regex-in-a-loop

**Review by houqp — CHANGES_REQUESTED** [review]

**Comment by houqp** [comment]
The rest looks good to me 👍 . I also filed https://github.com/delta-io/delta-rs/issues/93 as a follow up item.

**Comment by houqp** [comment]
Given that there is no mention of decimal type in the official delta spec: https://github.com/delta-io/delta/blob/master/PROTOCOL.md, could you share where see it being used?

**Comment by nevi-me** [comment]
> Given that there is no mention of decimal type in the official delta spec: https://github.com/delta-io/delta/blob/master/PROTOCOL.md, could you share where see it being used?

I think it's an omission on the document. I'll check this with a test file, and revert back.

**Comment by fvaleye** [comment]
@houqp I tested my branch (https://github.com/delta-io/delta-rs/pull/90) on our production Delta tables and an error was raised: 

```
exception:
thread '<unnamed>' panicked at 'unexpected delta schema type: decimal(20,2)', rust/src/delta_arrow.rs:62:25
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/florianvaleye/Documents/workspace/delta-rs/python/venv/lib/python3.9/site-packages/deltalake/__init__.py", line 43, in schema
    for field in self._table.schema(format=format.value).schema
pyo3_runtime.PanicException: unexpected delta schema type: decimal(20,2)

```

Example of a field with decimal type in the fields section of the delta table transaction:
`{"name":"price","type":"decimal(20,2)","nullable":true,"metadata":{}},`

**Comment by houqp** [comment]
awesome, in that case, @fvaleye you should also send a PR to https://github.com/delta-io/delta/blob/master/PROTOCOL.md :D

**Review by houqp — APPROVED** [review]

**Pull request merged**